client/jobs.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"google.golang.org/protobuf/types/known/anypb"
+	"google.golang.org/protobuf/types/known/durationpb"
+
+	commonpb "github.com/dapr/dapr/pkg/proto/common/v1"
+	runtimepb "github.com/dapr/dapr/pkg/proto/runtime/v1"
+)
+
+type FailurePolicy interface {
+	GetPBFailurePolicy() *commonpb.JobFailurePolicy
+}
+
+type JobFailurePolicyConstant struct {
+	maxRetries *uint32
+	interval   *time.Duration
+}
+
+func (f *JobFailurePolicyConstant) GetPBFailurePolicy() *commonpb.JobFailurePolicy {
+	policy := &commonpb.JobFailurePolicy{
+		Policy: &commonpb.JobFailurePolicy_Constant{
+			Constant: &commonpb.JobFailurePolicyConstant{},
+		},
+	}
+	if f.maxRetries != nil {
+		policy.Policy.(*commonpb.JobFailurePolicy_Constant).Constant.MaxRetries = f.maxRetries
+	}
+	if f.interval != nil {
+		policy.Policy.(*commonpb.JobFailurePolicy_Constant).Constant.Interval = &durationpb.Duration{Seconds: int64(f.interval.Seconds())}
+	}
+	return policy
+}
```

Comment:
Why and what is this?

Reply:
`JobFailurePolicyConstant` is the exposed type the SDK users can use to assign FailurePolicies. This function then converts this exposed type into the corresponding policy from the `commonpb` package.